express-notes-api/server.js

```diff
@@ -0,0 +1,89 @@
+import express from 'express';
+import * as fs from 'node:fs/promises';
+
+const app = express();
+
+const fileContent = await fs.readFile('data.json', 'utf-8');
+const data = JSON.parse(fileContent); // Parsed Object of data.json //
+
+const fileWrite = async () => {
+  await fs.writeFile('data.json', JSON.stringify(data, null, 2));
+};
+
+app.use(express.json()); // Incoming requests become parsed //
+
+app.get('/api/notes', (req, res) => {
+  const notesArr = Object.values(data.notes); // Inputting the object into an array //
+  res.status(200).send(notesArr);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const id = req.params.id;
+  if (id < 0) { // Checks to see if id is a positive integer //
+    res.status(400).send({ error: 'The id MUST be a positive integer!' });
+  } else if (!data.notes[id]) { // Checks to see if id exists //
+    res.status(404).send({ error: `The id:${id} was NOT found!` });
+  } else { // if id exists //
+    res.status(200).send(data.notes[id]);
+  }
+});
+
+app.post('/api/notes', async (req, res) => {
+  try {
+    if (!req.body.content) {
+      res.status(400).send({ error: 'Content is a required field' });
+    } else if (req.body.content) {
```

Comment:
The `if` part of the `else if` is redundant, since it's always `true` when the first `if` is `false`.